### PR TITLE
support tailx mode cleaner && fix ua config && fix socket reader panic

### PR DESCRIPTION
**Pull request by wonderflow**
## Changes
   
- [ ] feature1
- [ ] feature2
- [ ] fixbug1
- [ ] fixbug2
 
## Reviewers

- [ ] @[someone] please review
- [ ] @[someotherone] please review

## Wiki Changes
 
- options1...
- options2...
    
## Checklist
   
- [ ] Rebased/mergeable
- [ ] Tests pass
- [ ] Wiki updated

### CODE REVIEW

File: mgr/mgr.go [hunk 1]
```diff
@@ -356,23 +358,52 @@ func (m *Manager) handle(path string, watcher *fsnotify.Watcher) {
 	}
 }
 
+func (m *Manager) getCleanQueues(dir, file, mode string) ([]*cleanQueue, error) {
```
Comment (Unknwon) [inline mgr/mgr.go]: 既然是 Tailx 属于特判的话，我觉得先特判 `mode == reader.ModeTailx `，其它的作为默认处理方式，方便以后还要再加特判

File: mgr/mgr.go [hunk 2]
```diff
@@ -356,23 +358,52 @@ func (m *Manager) handle(path string, watcher *fsnotify.Watcher) {
...
+	if mode != reader.ModeTailx {
+		q, ok := m.cleanQueues[dir]
+		if !ok {
+			return nil, fmt.Errorf("cleaner dir %v not exist but got clean signal for delete file %v", dir, file)
+		}
+		return []*cleanQueue{q}, nil
+	}
+	var cleanQs []*cleanQueue
```
Comment (Unknwon) [inline mgr/mgr.go]: 这里直接命名 `cleanQueues` 更好点

File: mgr/mgr.go [hunk 3]
```diff
@@ -356,23 +358,52 @@ func (m *Manager) handle(path string, watcher *fsnotify.Watcher) {
...
-		log.Errorf("%v cleaner dir %v not exist but got clean signal for delete file %v", sig.Cleaner, dir, file)
+	queues, err := m.getCleanQueues(dir, file, sig.ReadMode)
+	if err != nil {
+		log.Error(sig.Cleaner, err)
 		return
 	}
-	count := q.filecount[file] + 1
-	if count >= q.cleanerCount {
+	//check if all queues can be cleaned
+	var canbedelete = true
```
Comment (Unknwon) [inline mgr/mgr.go]: s/canbedelete/canBeDeleted/

File: reader/meta.go [hunk 4]
```diff
@@ -187,6 +190,22 @@ func NewMetaWithConf(conf conf.MapConf) (meta *Meta, err error) {
...
+	if _, ok := m.subMetas[key]; ok {
+		return fmt.Errorf("subMeta %v is exist", key)
+	}
+	m.subMetas[key] = meta
+	return nil
+}
+
+func (m *Meta) RemoveSubMeta(key string) {
+	delete(m.subMetas, key)
+	return
```
Comment (Unknwon) [inline reader/meta.go]: 多余的 return

File: mgr/mgr.go (hunk 1 above)
Comment (wonderflow) [reply mgr/mgr.go]: ok

File: mgr/mgr.go (hunk 3 above)
Comment (wonderflow) [reply mgr/mgr.go]: hao

File: reader/meta.go (hunk 4 above)
Comment (wonderflow) [reply reader/meta.go]: hao

**Comment by Unknwon** [comment]
LGTM

**Pull request merged**